flink-connectors/flink-jdbc/src/main/java/org/apache/flink/api/java/io/jdbc/JDBCInputFormat.java

```diff
@@ -110,8 +116,9 @@
 	private String queryTemplate;
 	private int resultSetType;
 	private int resultSetConcurrency;
-	private RowTypeInfo rowTypeInfo;
+	private RowType rowType;
```

Comment:
InputFormat is a runtime concept, should we introduce the table ecosystem concept RowType in ?

Reply:
I haven't seen arguments to not use it in connectors. E.g. HiveTableInputFormat uses DataType. And it's much nicer for connector to just deal with user facing type system. What're your concerns?

Reply:
Conversions between `DataType` and `TypeInformation` is not lossless. It may introduce tricky issues.

Reply:
I agree. But I'm ok for the loss of some precision right now to keep up with the new type system. Once new type system is complete, we should be good to go